### PR TITLE
feat: add support for hetnzer cloud s3 storage

### PR DESCRIPTION
**Pull request by the-technat**
This adds support for Hetzner Cloud as provider for object storage.
 
Hcloud S3 is currently in beta and not yet officially announced. However there are already official [docs](https://docs.hetzner.com/storage/object-storage/) about it and it's listed as an entry in the Hcloud console. I received beta access this week and used their S3 together with this action [here](https://github.com/the-technat/tevbox/blob/main/.github/workflows/hcloud.yml#L67-L74). Since I had to workaround the missing provider I thought I'll try and contribute it here.

As region, Hetzner currently only provides `fsn1`.

Let me know what you think.

### CODE REVIEW

**Comment by bluebrown** [comment]
Hi, sorry for the delay. I think you need to precompile the thing to dist. `npm run build`.

**Comment by the-technat** [comment]
Oh thanks for the hint! Will do that soon.

**Comment by the-technat** [comment]
@bluebrown I run `npm run build` and commited the changes.

File: package.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
   "license": "ISC",
   "dependencies": {
     "@actions/core": "^1.2.6",
-    "@actions/github": "^4.0.0"
+    "@actions/github": "^4.0.0",
+    "ncc": "^0.3.6"
```
Comment (bluebrown) [inline package.json:15]: why did you need to add this one? Is the one from dev dependencies not working?

**Comment by bluebrown** [comment]
@the-technat , thank you for your contribution.
